platforms/jvm/plugins-java/src/integTest/groovy/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponentIntegrationTest.groovy

```diff
@@ -22,128 +22,6 @@ import org.gradle.integtests.fixtures.AbstractIntegrationSpec
  * Tests {@link DefaultJvmSoftwareComponent}.
  */
 class DefaultJvmSoftwareComponentIntegrationTest extends AbstractIntegrationSpec {
-
-    def "cannot instantiate component instances without the java base plugin applied"() {
```

Comment:
❌ We're deleting a fair bit of tests here. Why? 
These tests verify that we can create multiple components -- thus verifying that they are self-contained and independent of each other. 

IMO we want to continue to verify that multiple components can co-exist, even if in practice that is not what the java plugin does. We do not want the isolation between components and features to regress

Reply:
I removed these because it no longer makes sense to define a self-contained factory for a `Component`.  Components now require a `Feature` to be created first, and then be supplied to the component in the constructor.  

Expanding this factory to work would be require it to create Features first, which require `SourceSet`s and tasks to be created first, etc.  I got as far as:

```
registerFactory(JvmSoftwareComponentInternal) {
	SourceSet sourceSet = objects.newInstance(org.gradle.api.internal.tasks.DefaultSourceSet, it + 'SourceSet', objects)
    org.gradle.api.plugins.jvm.internal.DefaultJvmFeature feature = new org.gradle.api.plugins.jvm.internal.DefaultJvmFeature(it + 'Feature', sourceSet, [], project, false, false)
    objects.newInstance(DefaultJvmSoftwareComponent, it + 'Component', project, feature)
}
```

to replace it, then decided it wasn't worth it.

If the goal is to verify that multiple _components_ can co-exist, then we could maybe change all these to `DefaultAdhocSoftwareComponent` factories, or define a `TestComponent` or something, but that didn't seem to be the point of this test - they are explicitly about instantiating `DefaultJvmComponent`s.

After the changes here, this is much harder and these simple factory tests are no longer appropriate.

Reply:
> Components now require a Feature to be created first, and then be supplied to the component in the constructor.

I thought we had originally discussed an approach where a component has some container of features and that the plugin would add a feature to component's list of features. What happened to that plan?

Reply:
I was thinking of that as a longer term plan and a bigger change that what we wanted to do immediately here.

I'm still on board with it, and this doesn't prevent anything, but that wasn't a goal of mine here.

Reply:
I think we can achieve that with quite a small set of changes while also being able to preserve tests that verify multiple components existing simultaneously. The tests would only need to verify that two empty components can live side-by-side as long as we have separate tests elsewhere that verify multiple features live side-by-side.


I don't think its worth losing this test coverage over a change that can be applied in a couple of lines

Reply:
These tests aren't covering any behavior that a user can perform (without accessing internal types).  They aren't about components in general, they're specifically about `DefaultJvmSoftwareComponent`s, and they can be easily resurrected from the history when it does become possible to create multiple components.

I don't think this coverage is very valuable yet, and I don't want to expand the scope of this to support components containing a collection of features.  That's a bigger change that should be done separately.  This is a marginal improvement and a low risk step in the right direction, it's enough for one PR.